### PR TITLE
Added support to pass functions to index and llm complete functions

### PR DESCRIPTION
**Pull request by Deepak-Kesavan**
## What

Added support to pass functions to index and llm complete functions

## Why

- To process text after extraction
- To process text after LLM resoponse

## How

...

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions / Env Variables

-

## Notes on Testing

...

## Screenshots

...

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: src/unstract/sdk/index.py [hunk 1]
```diff
@@ -101,27 +104,6 @@ def query_index(
         finally:
             vector_db.close()
 
-    def _cleanup_text(self, full_text):
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/index.py:104]: @Deepak-Kesavan This function is not sued anywhere and hence removed?

File: src/unstract/sdk/llm.py [hunk 2]
```diff
@@ -69,15 +67,24 @@ def _initialise(self):
...
-        retries: int = 3,
+        process_text: Optional[Callable[[str], str]] = None,
         **kwargs: Any,
     ) -> Optional[dict[str, Any]]:
         try:
             response: CompletionResponse = self._llm_instance.complete(prompt, **kwargs)
+            process_text_output = {}
+            if process_text:
+                try:
+                    process_text_output = process_text(response, LLM.json_regex)
```
Comment (gaya3-zipstack) [inline src/unstract/sdk/llm.py:78]: What can be candidates for  process_text() @Deepak-Kesavan  ?

Comment (chandrasekharan-zipstack) [reply src/unstract/sdk/llm.py:78]: I suggest adding it as a docstring

Comment (Deepak-Kesavan) [reply src/unstract/sdk/llm.py:96]: Added docstring

File: src/unstract/sdk/llm.py [hunk 3]
```diff
@@ -68,15 +68,41 @@ def _initialise(self):
...
         prompt: str,
-        retries: int = 3,
+        process_text: Optional[Callable[[str], str]] = None,
         **kwargs: Any,
     ) -> Optional[dict[str, Any]]:
+        """Generates a completion response for the given prompt.
+
+        Args:
+            prompt (str): The input text prompt for generating the completion.
+            process_text (Optional[Callable[[str], str]], optional): A callable that
```
Comment (chandrasekharan-zipstack) [inline src/unstract/sdk/llm.py:78]: NIT: Try to also document what the arguments to the `process_text` can be. In the future if needed, we can allow the user to pass args for that function as well

**Review by chandrasekharan-zipstack — APPROVED** [review]
LGTM, please bump the minor version also

**Review by tahierhussain — APPROVED** [review]

File: src/unstract/sdk/llm.py (hunk 3 above)
Comment (Deepak-Kesavan) [reply src/unstract/sdk/llm.py:78]: Will take care of this later because it will be a internal use case for now.

**Pull request merged**